src/shady-css/token-cursor.js

```diff
@@ -0,0 +1,36 @@
+const index = Symbol('index');
+const lexer = Symbol('lexer');
+const nextToken = Symbol('nextToken');
+
+class TokenCursor {
```

Comment:
I'd think I'd much rather eliminate this class and add `peek` to Tokenizer. Where do you use the lookahead?